controllers/twophase/types.go

```diff
@@ -65,6 +67,29 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	}
 	chaos := _chaos.(v1alpha1.InnerSchedulerObject)
 
+	if chaos.GetStatus().Experiment.Phase == v1alpha1.ExperimentPhaseFinished {
+		// This chaos was finished, we should remove the finalizer
+		r.Log.Info("Removing pre-finalizer")
+
+		chaos.GetMeta().SetFinalizers(finalizer.RemoveFromFinalizer(chaos.GetMeta().GetFinalizers(), common.PreFinalizer))
+
+		updateError := r.Update(ctx, chaos)
```

Comment:
Why not handle conflict errors here?

Reply:
If it fails, it will be retried by the `controller-runtime` automatically, as pointed out by @Hexilee [here](https://github.com/chaos-mesh/chaos-mesh/pull/2509#discussion_r747281547)

Reply:
Got it!